public/app/plugins/datasource/prometheus/tracking.ts

```diff
@@ -0,0 +1,46 @@
+import { CoreApp, DataQueryRequest, DataQueryResponse } from '@grafana/data';
+import { reportInteraction, config } from '@grafana/runtime';
+
+import { PromQuery } from './types';
+
+export function trackQuery(
+  response: DataQueryResponse,
+  request: DataQueryRequest<PromQuery> & { targets: PromQuery[] },
+  startTime: Date
+): void {
+  const { app, targets: queries } = request;
+  // We do want to track panel-editor and explore
+  // We do not want to track queries from the dashboard or viewing a panel
+  // also included in the tracking is cloud-alerting, unified-alerting, and unknown
+  if (app === CoreApp.Dashboard || app === CoreApp.PanelViewer) {
+    return;
+  }
+
+  for (const query of queries) {
+    reportInteraction('grafana_prometheus_query_executed', {
+      app,
+      grafana_version: config.buildInfo.version,
+      has_data: response.data.some((frame) => frame.length > 0),
+      has_error: response.error !== undefined,
+      expr: query.expr,
```

Comment:
So this will essentially log every prometheus query string?

Reply:
Yep! This is modeled after the Loki query tracker and they are tracking lots more.